ttnn/cpp/ttnn/operations/conv/conv2d/conv2d_utils.cpp

```diff
@@ -848,8 +955,366 @@ void adjust_conv_op_config_for_auto_shard_if_necessary(
+        (conv_config.weights_dtype == DataType::BFLOAT8_B || conv_config.weights_dtype == DataType::BFLOAT16) &&
+        conv_config.output_layout == Layout::ROW_MAJOR && ((elem_size * in_channels) % (16 * num_cores_c)) == 0;
+    bool use_non_tile_height = (shard_layout == TensorMemoryLayout::HEIGHT_SHARDED) && out_channels <= 256 && conv_config.act_block_h_override == 0 &&
+        (conv_config.dtype == DataType::BFLOAT16 || conv_config.dtype == DataType::FLOAT32) && conv_config.output_layout == Layout::ROW_MAJOR && conv_config.input_channels_alignment != 16; //shalow conv varient
+
+    uint32_t round_up_size = !use_non_tile_height ? tt::constants::TILE_HEIGHT : 1;
+    uint32_t nhw_out = batch_size * output_height * output_width;
+    uint32_t out_channels_padded = tt::round_up(
+        out_channels,
+        get_num_cores_channels_from_parallel_config(output_parallel_config) * tt::constants::TILE_WIDTH);
+    if(is_non_tile_mul_width) {
+        out_channels_padded = tt::round_up(out_channels, 32);
+    }
+    MemoryConfig conv_out_memory_config = create_sharded_memory_config_from_parallel_config(
+        ttnn::Shape(std::array<uint32_t, 4>{1, 1, nhw_out, out_channels_padded}),
+        output_parallel_config,
+        round_up_size);
+    ParallelConfig largest_parallel_config = output_parallel_config.grid.num_cores() > input_parallel_config.grid.num_cores() ? output_parallel_config : input_parallel_config;
+
+    OptimizedConvParallelizationConfig opt_conv_op_parallel_config = determine_conv_op_parallel_config_from_conv_output_mem_config(
+        conv_out_memory_config,
+        get_num_cores_nhw_from_parallel_config(largest_parallel_config),
+        get_num_cores_channels_from_parallel_config(largest_parallel_config));
+
+    uint32_t in_channels_padded = tt::round_up(
+        in_channels,
+        get_num_cores_channels_from_parallel_config(input_parallel_config) * conv_config.input_channels_alignment);
+    if(is_non_tile_mul_width){
+        in_channels_padded = tt::round_up(in_channels, conv_config.input_channels_alignment);
+    }
+
+    uint32_t nhw_out_padded_ntile = get_num_cores_nhw_from_parallel_config(output_parallel_config) *
+                                    conv_out_memory_config.shard_spec.value().shape[0] / tt::constants::TILE_HEIGHT;
+
+    OptimizedConvBlockConfig opt_conv_op_block_config = determine_per_core_conv_block_config(
+        input_parallel_config,
+        opt_conv_op_parallel_config,
+        in_channels_padded,
+        nhw_out_padded_ntile,
+        conv_config.act_block_h_override,
+        conv_config.act_block_w_div,
+        kernel_size[0],
+        kernel_size[1],
+        get_fp32_dest_acc_en(compute_config),
+        conv_config.enable_split_reader);
+    return {opt_conv_op_parallel_config, opt_conv_op_block_config, conv_out_memory_config};
+}
+
+std::pair<uint32_t,uint32_t> conv2d::estimate_L1_usage(
```

Comment:
are cbs for sharded inputs and outputs accounter here as well or just tmp CBs?

Reply:
CBs that have globally allocated addresses, like the input & output are not accounted for in the CB size estimation.  Only CBs that need memory allocation via the program object are accounted for.

Reply:
You should document this on function header (what it does and it doesn't do)

Reply:
Done.